### ISSUE TITLE
Support dot dot in pearlite

### ISSUE THREAD

**Issue by bclement-ocp** [xldenis/creusot#768]
Sadly, pearlite doesn't seem to support the "dot dot" or "with" operator.

```rust
use creusot_contracts::{logic::FMap, logic::FSet, *};

#[derive(Debug)]
struct A { l: usize, r: usize }

impl A {
    #[logic]
    fn with_l(self, l: usize) -> Self {
        A { l, .. self }
    }
}
```

Running why3:
```
File "../proto-rlib.mlcfg", line 169, characters 4-50: This term has type mach.int.UInt64Gen.uint64 -> Proto_A_Type.t_a,
but is expected to have type Proto_A_Type.t_a
```

**Comment by xldenis** [comment]
Given the error message it seems like the parser, and type checker are correctly handling it but the lowering to Why3 ignores the `..X` part.

**Comment by bclement-ocp** [comment]
I still get the error with the fix from #769 :(

I should have specified the error occurs from inside why3, not when running creusot.
